python/kvikio/nvcomp_codec.py

```diff
@@ -95,28 +95,16 @@ def encode_batch(self, bufs: List[Any]) -> List[Any]:
         comp_chunk_size = self._algo.get_compress_chunk_size(max_chunk_size)
 
         # Prepare data and size buffers.
-        uncomp_chunks = cp.array(
-            [b.data.ptr for b in bufs],
-            dtype=cp.uint64,
-        )
+        uncomp_chunks = cp.array([b.data.ptr for b in bufs], dtype=cp.uint64)
         uncomp_chunk_sizes = cp.array(buf_sizes, dtype=cp.uint64)
 
         temp_buf = cp.empty(temp_size, dtype=cp.uint8)
 
-        # Includes header with the original buffer size,
-        # same as in numcodecs codec. This enables data compatibility between
-        # numcodecs default codecs and this nvCOMP batch codec.
-        # TODO(akamenev): probably should use contiguous buffer which stores all chunks?
-        comp_chunks_header = [
-            cp.empty(self.HEADER_SIZE_BYTES + comp_chunk_size, dtype=cp.uint8)
-            for _ in range(num_chunks)
-        ]
-        # comp_chunks is used as a container that stores pointers to actual chunks.
+        comp_chunks = cp.empty((num_chunks, comp_chunk_size), dtype=cp.uint8)
```

Comment:
OK, so you allocate a single contiguous piece of memory for the compressed chunks and then pass offset pointers to nvcomp...

Reply:
Yes, I decided to allocate one chunk of memory, reducing the number of allocations. The potential downside is that if there is a memory pressure and fragmentation, the allocator might not find large enough chunk while with smaller, non-contiguous chunks it might succeed.